wweval/R/score_real_time_outputs.R

```diff
@@ -0,0 +1,289 @@
+                                    dates,
+                                    eval_data,
+                                    hosp_only = FALSE) {
+  if (isTRUE(hosp_only)) {
+    model_types <- c("hosp")
+  } else {
+    model_types <- c("ww", "hosp")
+  }
+  all_scores <- c()
+
+  data_type <- ifelse(score_type == "crps", "draws", "quantiles")
+  col_name <- ifelse(score_type == "crps", "draw", "quantile")
+  for (i in seq_along(dates)) {
+    date_to_pull <- dates[i]
+    metadata <- table_of_run_ids |> dplyr::filter(
+      forecast_date == date_to_pull
+    )
+    run_id <- metadata$ids
+    date_run <- metadata$dates_run
+    for (j in seq_along(locations)) {
+      for (m in seq_along(model_types)) {
+        model_long <- ifelse(model_types[m] == "ww",
+          "site-level infection dynamics",
+          "hospital admissions only"
+        )
+        if (date_to_pull %in% c("2024-02-05", "2024-02-12")) {
+          # Assume the old  file structure
+          fp <- file.path(
+            real_time_output_dir,
+            glue::glue("output_{date_to_pull}"),
+            "raw",
+            locations[j],
+            model_long,
+            data_type,
+            date_to_pull,
+            glue::glue("run-on-{date_run}-{run_id}-{data_type}.parquet")
+          )
+          if (file.exists(fp)) {
+            # The diagnostics are not flags here, just values.
+            any_flags <- FALSE
+
+            these_preds <- arrow::read_parquet(fp) |>
+              dplyr::filter(
+                name == "pred_hosp",
+                period != "calibration"
+              ) |>
+              dplyr::select(
+                forecast_date,
+                date,
+                location,
+                value,
+                !!sym(col_name)
+              ) |>
+              dplyr::mutate(
+                model = model_types[m],
+                failed_convergence = any_flags
+              )
+          } else {
+            these_preds <- c()
+          }
+        } else {
+          # Assume the main newer file structure
+          dir <- file.path(
+            real_time_output_dir,
+            date_to_pull,
+            glue::glue("run-on-{date_run}-{run_id}"),
+            "raw",
+            locations[j],
+            model_long
+          )
+
+          if (file.exists(file.path(dir, glue::glue("{data_type}.parquet")))) {
+            this_flags <- readr::read_csv(file.path(dir, "diagnostics.csv"))
+            any_flags <- any(this_flags$value[20:23] == TRUE)
+
+            these_preds <- arrow::read_parquet(
+              file.path(dir, glue::glue("{data_type}.parquet"))
+            ) |>
+              dplyr::filter(
+                name == "pred_hosp",
+                period != "calibration"
+              ) |>
+              dplyr::select(
+                forecast_date,
+                date,
+                location,
+                value,
+                !!sym(col_name)
```

Comment:
Don't need the `sym`
```suggestion
                !!col_name
```